pkg/k8s/version/version.go

```diff
@@ -164,6 +185,72 @@ func fallbackDiscovery(client kubernetes.Interface) error {
 	return fmt.Errorf("unable to validate EndpointSlices support: %s", err)
 }
 
+func leasesFallbackDiscovery(client kubernetes.Interface) error {
+	cached.mutex.Lock()
+	// Here we check if cilium has authorization to check for leases object or not.
+	// Prior to v1.14 leases.coordination.k8s.io/v1 was not supported and hence cilium
+	// ClusterRole did not had access to corresponding API resource.
+	// Look at the helm manifest to check the ClusterRole associated with cilium Deployments
+	// corresponding to the configured Kubernetes version.
+	if cached.version.Minor < 14 && cached.version.Major == 1 {
+		cached.mutex.Unlock()
+		return nil
+	}
+
+	_, err := client.CoordinationV1().Leases("kube-system").Get(context.TODO(), "kube-controller-manager", metav1.GetOptions{})
+	if err == nil {
+		cached.mutex.Lock()
+		cached.capabilities.LeasesResourceLock = true
+		cached.mutex.Unlock()
+		return nil
+	}
+
+	switch t := err.(type) {
```

Comment:
There is a k8s helper function to check for error not found.
https://github.com/cilium/cilium/blob/f694b3ba28ac2936c413fddec7771ef14a2c8808/vendor/k8s.io/apimachinery/pkg/api/errors/errors.go#L485-L488